tests/vmi_multus_test.go

```diff
@@ -846,6 +848,45 @@ var _ = Describe("SRIOV", func() {
 			// it's hard to match them.
 		})
 
+		Context("VLANed SR-IOV network", func() {
+			var vlanedVMI1 *v1.VirtualMachineInstance
+			cidrVlaned1 := "192.168.0.1/24"
+			cidrVlaned2 := "192.168.0.2/24"
+			cidrFlat := "192.168.1.1/24"
+			BeforeEach(func() {
+				createNetwork("sriov-vlan", tests.NamespaceTestDefault, sriovConfVlanCRD)
+				createNetwork("sriov-no-vlan", tests.NamespaceTestDefault, sriovConfNoVlanCrd)
+				vlanedVMI1 = createSriovVMI([]string{"sriov-vlan"})
+
+				// manually configure IP/link on sriov interfaces because there is
+				// no DHCP server to serve the address to the guest
+				configInterface(vlanedVMI1, "eth1", cidrVlaned1, "#")
+			})
+
+			AfterEach(func() {
+				deleteNetwork("sriov-vlan", tests.NamespaceTestDefault)
+				deleteNetwork("sriov-no-vlan", tests.NamespaceTestDefault)
+			})
```

Comment:
What about deleting the vmi?

Reply:
Currently we get errors from testSuite because the test namesapces wont delete and timed out, 
lets add deleting VMI to before each as well

Reply:
@ormergi I think all the VMIs in the tests namespace are deleted in the beforeEach()

Reply:
Done.

Reply:
You need to check is `vlanedVMI1` is not `nil`, because its creation may not have occurred (due to failures).

There is also the matter of waiting or not waiting for the deletion to finish (as this is an async operation).
I do not have a definitive answer here for the VMI, mainly because its name is randomized so the next test should not collide. Not sure about the network objects.

Reply:
As Alon mentioned. There is [BeforeEach](https://github.com/kubevirt/kubevirt/pull/3479/files#diff-897951be21489aebaa1df0c66894562cR614) that is taking care for the vm cleanup including waiting for it to disappear. It is not a generic cleanup. It is here, in this file, under the SRIOV describe section. Plus as Eddy mentioned the vm name is randomized, so the next test shouldn't collide. For those reasons, I think the vm cleanup here is completely redundant and even confusing. The next developer that with touch those tests won't be able to understand why this specific test has a special double cleanup for each vm, while the others don't.

Regarding the newly introduced `DeleteVMI`, nice. I would use it in the mentioned `BeforeEach` "general" cleanup.

Reply:
I am also in favor of the "general" cleanup because I also think it is confusing.
If we are totally against this "general" cleanup then, I can followup with another PR to remove the "general" cleanup from all cases, and make each test cleanup It's own objects. I don't think this is necessary though. 
For this PR I think relaying on the before each with the changes Alona purposed for the `BeforeEach` should be enough.
@EdDev  what do you think?

Reply:
The errors that we get from test suite I mentioned are errors from AfterSuite, they are not related to VMI's that wont delete like I though it was something else, it is explained in this issue #3850 

I think the problem here is that when SRIOV context tests finish, there are VMI's left in the cluster.
Since there is no `AfterAll` method we need to delete all VMI's at after each test so there will be no VMI left in the cluster after the last test is finished.
I think you should move this `AfterEach` block to the top level of sriov tests context (line 570)

Reply:
Anyway, I don't want to block this PR that anyway has long history. If you believe cleaning the vms here as well is a better practice, I don't mind.

Reply:
Maybe to avoid redundant  vmi-cleanups,  (let's say _test2_ runs after _test1_, and there is a clean up after _test1_,  so a redundant vmi cleanup will be before _test2_) I make a clean up in the `BeforeAll()`  and add an `AfterEach()` block that will cleanup after each case. 
Sounds good?

Reply:
There is already a cleanup of VMI's (and other objects) at `BeforeAll()` https://github.com/kubevirt/kubevirt/blob/28188d3d0c4c2eb940e9259ad8873a52e702fa2f/tests/vmi_multus_test.go#L600
so If you want to ensure there are no VMI before and after each test, you should move this `AfterEach` block to the top level SRIOV context.

Reply:
@alonSadan the suggested follow-up PR sounds reasonable to me.

Reply:
When working on the follow-up PR please consider using `DeleteAndWaitForVirtualMachineToDisappear` in the `BeforeAll` block as well.

Reply:
> There is already a cleanup of VMI's (and other objects) at `BeforeAll()`
> 
> https://github.com/kubevirt/kubevirt/blob/28188d3d0c4c2eb940e9259ad8873a52e702fa2f/tests/vmi_multus_test.go#L600
> 
> 
> so If you want to ensure there are no VMI before and after each test, you should move this `AfterEach` block to the top level SRIOV context.

Thanks. will make it in the followup

Reply:
> @alonSadan the suggested follow-up PR sounds reasonable to me.

great. so I am living it intact for now